starboard/nplb/maximum_player_configuration_explorer_test.cc

```diff
@@ -0,0 +1,117 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <set>
+#include <string>
+#include <vector>
+
+#include "starboard/nplb/drm_helpers.h"
+#include "starboard/nplb/maximum_player_configuration_explorer.h"
+#include "starboard/nplb/player_test_util.h"
+#include "starboard/shared/starboard/player/video_dmp_reader.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace starboard {
+namespace nplb {
+namespace {
+
+using shared::starboard::player::video_dmp::VideoDmpReader;
+using ::testing::Combine;
+using ::testing::ValuesIn;
+
+TEST(SearchPosetMaximalElementsTests, SearchPosetMaximalElementsTest) {
+  constexpr int kResource_types = 3;
+  constexpr int kMaxInstances = 5;
+
+  std::set<std::vector<int>> test_pattern_set = {
+      {1, 4, 5}, {2, 5, 3}, {3, 3, 4}, {3, 4, 3}, {4, 0, 3}, {5, 5, 2}};
+
+  auto test_funtion = std::bind(PosetContainedByMaxElementsSet,
+                                test_pattern_set, std::placeholders::_1);
+
+  auto max_test_config_set = SearchPosetMaximalElementsDFS(
+      kResource_types, kMaxInstances, test_funtion);
+
+  EXPECT_EQ(max_test_config_set, test_pattern_set);
+}
+
+class MaximumPlayerConfigurationExplorerTest
+    : public ::testing::TestWithParam<
+          ::testing::tuple<SbPlayerOutputMode, const char*>> {
+ protected:
+  MaximumPlayerConfigurationExplorerTest() {}
+
+  testing::FakeGraphicsContextProvider fake_graphics_context_provider_;
+};
+
+TEST_P(MaximumPlayerConfigurationExplorerTest, SunnyDay) {
+  const int kMaxPlayerInstancesPerConfig = 5;
+  const int kMaxTotalPlayerInstances = 5;
+
+  SbPlayerOutputMode output_mode = std::get<0>(GetParam());
+  const char* key_system = std::get<1>(GetParam());
+
+  const std::vector<const char*>& video_test_files = GetVideoTestFiles();
+  std::vector<SbPlayerTestConfig> supported_player_configs;
+  for (const auto& video_filename : video_test_files) {
+    VideoDmpReader dmp_reader(video_filename,
+                              VideoDmpReader::kEnableReadOnDemand);
+    SB_DCHECK(dmp_reader.number_of_video_buffers() > 0);
+    if (SbMediaCanPlayMimeAndKeySystem(dmp_reader.video_mime_type().c_str(),
+                                       key_system)) {
+      supported_player_configs.emplace_back(nullptr, video_filename,
+                                            output_mode, key_system);
+    }
+  }
+
+  if (supported_player_configs.empty()) {
+    // TODO: Use GTEST_SKIP when we have a newer version of gtest.
+    SB_LOG(INFO) << "No supported video codec. Skip the tests.";
+    return;
+  }
+
+  MaximumPlayerConfigurationExplorer explorer(
```

Comment:
Is this purpose of this test just to make sure it doesn't crash? I think we should add a comment somewhere explaining why there's no assert or expect statements.